File: languages/python/bloock/entity/authenticity/signer_args.py
from bloock.entity.key.local_key import LocalKey
from bloock.entity.key.managed_key import ManagedKey
from bloock.entity.key.managed_certificate import ManagedCertificate
from bloock.entity.key.local_certificate import LocalCertificate


class SignerArgs:
    local_key = None
    managed_key = None
    managed_certificate = None
    local_certificate = None

    def __init__(self, key, common_name=None) -> None:
        if isinstance(key, LocalKey):
            self.local_key = key
        elif isinstance(key, ManagedKey):
            self.managed_key = key
        elif isinstance(key, ManagedCertificate):
            self.managed_certificate = key
        elif isinstance(key, LocalCertificate):
            self.local_certificate = key
        else:
            raise Exception(
                "Invalid key provided. Must be of type LocalKey or ManagedKey")

        self.common_name = common_name

File: languages/php/Bloock/Entity/Authenticity/SignerArgs.php
<?php

namespace Bloock\Entity\Authenticity;

use Bloock\Entity\Key\LocalKey;
use Bloock\Entity\Key\ManagedCertificate;
use Bloock\Entity\Key\ManagedKey;
use Exception;

class SignerArgs
{
    public ?LocalKey $localKey = null;
    public ?ManagedKey $managedKey = null;
    public ?ManagedCertificate $managedCertificate = null;
    public string $commonName;

    /**
     * @throws Exception
     */
    public function __construct($key, string $commonName = "")
    {
        if ($key instanceof LocalKey) {
            $this->localKey = $key;
        } else if ($key instanceof ManagedKey) {
            $this->managedKey = $key;
        } else if ($key instanceof ManagedCertificate) {
            $this->managedCertificate = $key;
        } else {
            throw new Exception("Invalid $key provided. Must be of type LocalKey or ManagedKey");
        }

        $this->commonName = $commonName;
    }
}
